src/shared/components/Settings/Preferences/Email/index.jsx

```diff
@@ -93,11 +93,20 @@ export default class EmailPreferences extends ConsentComponent {
     }
   }
 
+  shouldComponentUpdate(nextProps) {
+    const { profileState: { emailPreferences } } = nextProps;
+    if (emailPreferences && this.props.profileState.emailPreferences !== emailPreferences) {
+      return true;
+    }
+    return false;
+  }
+
   onHandleChange(id, checked) {
     this.showConsent(this.onChange.bind(this, id, checked));
   }
 
   onChange(id, checked) {
+    document.querySelectorAll(`#pre-onoffswitch-${id}`).forEach((el) => { el.checked = checked; }); // eslint-disable-line no-param-reassign
```

Comment:
@ThomasKranitsas this line was required, needed to mark it check/uncheck explicitly. good one

Reply:
Well, we always need to find workarounds as a developer. 😉